public/app/features/live/live.ts

```diff
@@ -95,6 +93,77 @@ class CentrifugeSrv implements GrafanaLiveSrv {
     console.log('onError', context);
   };
 
+  /**
+   * Returns a channel that has been initalized and subscribed to the given path
+   */
+  private async getChannel<T>(pluginId: string, path: string): Promise<Channel<T>> {
+    const key = `${pluginId}/${path}`;
+    const c = this.channels.get(key);
+    if (c) {
+      return Promise.resolve(c);
+    }
+
+    const liveSupport = await this.getChannelSupport(pluginId);
+    const handler = liveSupport.getChannelHandler(path);
+    if (!handler) {
+      return Promise.reject(`Invalid path: ${key}`);
+    }
+
+    // Make sure we are connected before trying to subscribe
+    if (!this.isConnected()) {
```

Comment:
@kaydelaney -- I copied some stuff from #27600

This should have similar behavior now :crossed_fingers: